include/circt/Dialect/SMT/SMTOps.td

```diff
@@ -49,6 +49,25 @@ def DeclareConstOp : SMTOp<"declare_const", [
   }];
 }
 
+def BoolConstantOp : SMTOp<"constant", [
+  Pure,
+  ConstantLike,
+  DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
+]> {
+  let summary = "Produce a constant boolean";
+  let description = [{
+    Produces the constant expressions 'true' and 'false' as described in the
+    [Core theory](https://smtlib.cs.uiowa.edu/Theories/Core.smt2) of the SMT-LIB
+    Standard 2.6.
+  }];
+
+  let arguments = (ins BoolAttr:$value);
+  let results = (outs BoolType:$result);
+  let assemblyFormat = "$value attr-dict";
+
```

Comment:
You'll probably at some point want a builder that just take a boolean, but maybe that's already generated (?)

Reply:
Had the same thought, so I checked, and it's actually already auto-generated 👍